src/components/orderArea/OrderArea.jsx

```diff
@@ -1,73 +1,23 @@
-import React, { useState, useContext, useEffect, useReducer } from 'react';
+import React, { useState } from 'react';
 import { Container, BtnWrap, ReturnBtn } from 'components/orderArea/OrderArea.style';
-import { SelectedProductContext } from 'App';
-import { FinalPayContext } from 'pages/VendingMachine';
 import MoneySlot from 'components/orderArea/MoneySlot';
 import PutBtn from 'components/orderArea/PutBtn';
 import History from 'components/orderArea/History';
-import ProductHoll from 'components/orderArea/ProductHoll';
-
-const TIME_TO_PUT_OUT_PRODUCT = 2000;
-const TIME_TO_SELCT_PRODUCT = 5000;
-
-const historyReducer = (historyList, { action, newHistoryState }) => {
-  switch (action) {
-    case 'INPUT':
-      return [...historyList, `💸 총 ${newHistoryState}원을 투입하였습니다.`];
-    case 'SELECT':
-      return [...historyList, `${newHistoryState} 선택하였습니다.`];
-    default:
-      return historyList;
-  }
-};
+import ProductHole from 'components/orderArea/ProductHole';
 
 export default function OrderArea() {
   const useInputPayState = useState('');
-  const [canOrderState, setOrderState] = useState(true);
-  const [historyList, dispatchHistoryList] = useReducer(historyReducer, []);
-  const [selectedProduct, setSelectedProduct] = useContext(SelectedProductContext);
-  const finalPay = useContext(FinalPayContext)[0];
-
-  let timerIDToSelectProduct;
-  const startTimerToSelectProduct = () => {
-    timerIDToSelectProduct = setTimeout(() => {}, TIME_TO_SELCT_PRODUCT);
-  };
-
-  const initVMState = () => {
-    setSelectedProduct({});
-    setOrderState(true);
-  };
-
-  const startTimerToPutOutProduct = () => {
-    setTimeout(initVMState, TIME_TO_PUT_OUT_PRODUCT);
-  };
-
-  const putOutProduct = () => {
-    clearTimeout(timerIDToSelectProduct);
-    setOrderState(false);
-    startTimerToPutOutProduct();
-  };
-
-  useEffect(() => {
-    if (selectedProduct.detail) {
-      putOutProduct();
-      dispatchHistoryList({ action: 'SELECT', newHistoryState: selectedProduct.detail });
-    } // else dispatchHistoryList(); // TODO
-  }, [selectedProduct]);
-
-  useEffect(() => {
-    if (finalPay) dispatchHistoryList({ action: 'INPUT', newHistoryState: finalPay.toLocaleString('en') });
-  }, [finalPay]);
+  const [canOrderState] = useState(true);
 
   return (
     <Container>
       <MoneySlot useInputPayState={useInputPayState} canOrderState={canOrderState} />
       <BtnWrap>
-        <PutBtn inputPay={useInputPayState[0]} startTimerToSelectProduct={startTimerToSelectProduct} />
+        <PutBtn inputPay={useInputPayState[0]} />
```

Comment:
destructruring해서 넘겨주면 하위에서 이해하기 더 용이할 것 같네요 
```suggestion
        <MoneySlot useInputPayState={inputPay} onChange={setInputPay}canOrderState />
        <PutBtn inputPay={inputPay} />
```

   const [canOrderState] = useState(true); -> 이렇게 쓰면 true값만 넘어가지 않나요?

Reply:
setOderState를 사용하는 로직을 아직 추가 안해서 우선 지워뒀습니다!
추가한 부분 다음 PR때 보내도록하겠습니다 :)